pr47-core/src/vm/al31f/executor/async_executor.rs

```diff
@@ -539,8 +541,8 @@ async unsafe fn vm_thread_run_function_impl<A: Alloc>(
                 }
                 ffi_args.set_len(args.len());
 
-                let mut combustor: Combustor<A> = Combustor::new(
-                    NonNull::from(&thread.vm)
+                let mut combustor: AsyncCombustor<A> = AsyncCombustor::new(
+                    thread.vm.derive_child_serializer_no_task()
```

Comment:
This change is important. However, by this time there's no way to "check" (either at compile time, or at run time) if a serializer has a permission, making our system hard to use and unsound.
@chuigda @SchrodingerZhu

Reply:
Nice, fine, give it one more!